plugins/woocommerce/legacy/js/frontend/single-product.js

```diff
@@ -93,13 +93,16 @@ jQuery( function( $ ) {
 
 		// No images? Abort.
 		if ( 0 === this.$images.length ) {
-			this.$target.css( 'opacity', 1 );
 			return;
 		}
 
 		// Make this object available.
 		$target.data( 'product_gallery', this );
 
+		if ( this.$target.css( 'opacity' ) ) {
```

Comment:
Can you confirm expectations here/should we be testing if the `opacity` is `< 1` (the condition will be satisfied when `opacity` is already `1` ... but in that case it seems pointless to follow up with a further call setting it to `1`)?

Reply:
at the very beginning of this with @Mte90, we decided to leave that for backward compatibility, because the [single-product/product-image.php](https://github.com/woocommerce/woocommerce/pull/33041/files#diff-723f07f585c4e7d7da9112b69a0b5e258813b897cc7cc9a7f7f671a07f9f6e4eL38-L40) maybe unchanged

Reply:
I'm still a little unclear; if this condition is met we will set `opacity` to `1`. So, I'm guessing we only want to do that if `opacity` is not already `1`?

It doesn't really matter, in a way, because in that case nothing changes—my point is more that we can save ourselves an unnecessary call.

Reply:
yep sorry I wasn't clear! Some users could still have the [old template which specify opacity:0 with a style](https://github.com/woocommerce/woocommerce/pull/33041/files#diff-723f07f585c4e7d7da9112b69a0b5e258813b897cc7cc9a7f7f671a07f9f6e4eL38-L40) so for that the reason in order to avoid missing images we left that. Anyway it could be removed, with the new version of the template there would be no problem.